src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -227,6 +226,7 @@ public bool TryRemoveContext(IAzureContext context)
                 }
             }
 
+            bool shouldPopulateContextList = _profile.DefaultContext?.Account == null;
```

Comment:
Just in case, we should also ensure that _profile?.Contexts?.Count < 2 (actually, this will need to be two null checks, but you get the idea).

Reply:
@markcowl in this class, `_profile` is of the abstracted type `IProfileOperations`, which doesn't contain the `Contexts` property (the implementation `AzureRmProfile` does contain this property, but I doubt we want to use that in this class). Is there a separate check we can do here?

Reply:
makes sense. As long as we are not overwriting existing contexts, I think we are ok here